tests/scan.cc

```diff
@@ -286,3 +286,152 @@ TEST(scan_depth_zero) {
   }
   TRY(mapper.max_seen_scan_depth() == 0);
 }
+
+#define PERM_r CHERI_PERM_LOAD
+#define PERM_w CHERI_PERM_STORE
+#define PERM_x CHERI_PERM_EXECUTE
+#define PERM_rR (CHERI_PERM_LOAD | CHERI_PERM_LOAD_CAP)
+#define PERM_wW (CHERI_PERM_STORE | CHERI_PERM_STORE_CAP)
+
+void* __capability cap_with_perms(size_t size, size_t perms) {
+  void* ptr = calloc(1, size);
+#ifdef __CHERI_PURE_CAPABILITY__
+  void* cap = ptr;
+#else
+  void* __capability cap = cheri_address_set(cheri_ddc_get(), (ptraddr_t)ptr);
+  cap = cheri_bounds_set_exact(cap, 8);
+#endif
+  return cheri_perms_and(cap, perms);
+}
+
+#include <sys/sysctl.h>
+
+#ifdef __CHERI_PURE_CAPABILITY__
+#define CAP_TO(x) (&(x))
+#else
+#define CAP_TO(x) \
+  (cheri_bounds_set_exact(cheri_address_set(cheri_ddc_get(), (ptraddr_t) & (x)), sizeof(x)))
+#endif
```

Comment:
I think this is basically `cap(x)`. This one is exact, but I don't think it's important for the tests.

Reply:
At the [top of tests/scan.cc](https://github.com/capablevms/compartment-mapper/commit/aaa9ef82992f24c455fd5309ddca6cb115365c80#diff-9cc786caf774c2d148ea65d19a120274aaca6fa1ecd447323c53cd1371b13c71R21).

Reply:
I've switched to using `cap(x)` now